server/controllers/applicationsController.ts

```diff
@@ -174,10 +190,49 @@ const updateApplication = async (
   }
 };
 
+const getAggregatedUserStats = async (
+  req: CustomRequest<{ userId: string }>,
+  res: Response,
+  next: NextFunction
+) => {
+  const { userId } = req.params;
```

Comment:
do we need to send the userId as a parameter? Why not just use the id from the decoded jwt?

Reply:
SImilar reasoning to this https://github.com/Code-Hammers/code-hammers/pull/127#discussion_r1626559336

this route will eventually not just get the current users data but eventually would be able to fetch other peoples data that was made public.

Reply:
In that case we shouldn't do the check on line 199 right? That will only allow a user to see their own stats

Reply:
It's a temp protection to make sure that just because someone has a valid cookie (which the protect route ensures) they can't ping the route with another users ID. Once the functionality is built out to allow other users to use this route to check other peoples ID then the check on 199 would be replace with an isPublic check..

Reply:
works for me. hulk merge!

Reply:
👹